src/foam/nanos/notification/email/GroupEmailTemplateService.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.notification.email',
+  name: 'GroupEmailTemplateService',
+
+  documentation: 'Used in conjuction with ChainedPropertyService',
+
+  implements: [
+    'foam.nanos.notification.email.EmailPropertyService'
+  ],
+
+  javaImports: [
+    'foam.dao.DAO',
+    'foam.nanos.auth.Group',
+    'foam.nanos.logger.Logger',
+    'foam.util.SafetyUtil',
+  ],
+
+  methods: [
+    {
+      name: 'apply',
+      type: 'foam.nanos.notification.email.EmailMessage',
+      args: [
+        {
+          name: 'x',
+          type: 'Context',
+        },
+        {
+          name: 'group',
+          class: 'String',
+          documentation: 'group of user whose the recipient of the email being sent'
+        },
+        {
+          name: 'emailMessage',
+          type: 'foam.nanos.notification.email.EmailMessage',
+          documentation: 'Email message'
+        },
+        {
+          name: 'templateArgs',
+          type: 'Map',
+          documentation: 'Template arguments'
+        }
+      ],
+      documentation:
+      `Looks for group email properties on passed in group and then up the chain to root group.
+       Note: It is possible that only one of the properties is set on the group.`,
+      javaCode: `
+        Group grp = null;
+        do {
+          grp = (Group)((DAO) x.get("groupDAO")).find(group);
+          if ( grp == null ) return emailMessage;
+          group = grp.getParent();
+        } while ( SafetyUtil.isEmpty(grp.getReplyTo()) && SafetyUtil.isEmpty(grp.getDisplayName()) );
```

Comment:
Do we ONLY want to use the replyTo of a group if the displayName is set?
For example if I have:
Group A - replyTo: "group@mail.com" and displayName: "Group A"
Group B - child of group A, displayName: "Group B"

What should the behaviour be? I think I'd expect it to be an email from "Group B <group@mail.com>".

Reply:
Ya to be honest I pondered this too. Glad you caught it. Ok so my thoughts are that if we are setting the values on a group then we are customizing this group and thus would be setting the values we want on the group.

But now that I'm thinking about it again, if I change one property to be customized, I'd expect the other properties to remain the same.

I'll make these changes. With next push